src/server.ts

```diff
@@ -16,51 +10,44 @@ import {
   createRequestHandler as createRemixRequestHandler
 } from '@remix-run/node'
 
-import { createRemixAdapter } from './adapters'
-
-export enum AWSProxy {
-  APIGatewayV1 = 'APIGatewayV1',
-  APIGatewayV2 = 'APIGatewayV2',
-  ALB = 'ALB'
+export interface RemixAdapter<T, U> {
+  createRemixRequest: (event: T) => NodeRequest
+  sendRemixResponse: (nodeResponse: NodeResponse) => Promise<U>
 }
-
 /**
  * A function that returns the value to use as `context` in route `loader` and
  * `action` functions.
  *
  * You can think of this as an escape hatch that allows you to pass
  * environment/platform-specific values through to your loader/action.
  */
-export type GetLoadContextFunction = (
-  event: APIGatewayProxyEventV2 | APIGatewayProxyEvent | ALBEvent
+export type GetLoadContextFunction = <T>(
+  event: T
 ) => AppLoadContext;
 
-export type RequestHandler = APIGatewayProxyHandlerV2 | APIGatewayProxyHandler | ALBHandler;
-
 /**
  * Returns a request handler for Architect that serves the response using
  * Remix.
  */
-export function createRequestHandler({
+export function createRequestHandler<T, U>({
   build,
   getLoadContext,
   mode = process.env.NODE_ENV,
-  awsProxy = AWSProxy.APIGatewayV2
+  adapter,
 }: {
   build: ServerBuild;
   getLoadContext?: GetLoadContextFunction;
   mode?: string;
-  awsProxy?: AWSProxy;
-}): RequestHandler {
+  adapter: RemixAdapter<T, U>, 
```

Comment:
good point, devs would even be able to wrap the existing adapter and use it here